components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/RestApiAdminUtils.java

```diff
@@ -156,35 +164,62 @@ public static String constructMissingThrottleObjectErrorMessage(Enum<?> typeEnum
     }
 
     /**
-     * Extract the content of the provided tenant theme archive
+     * Import the content of the provided tenant theme archive to the file system and the database
      *
-     * @param themeFile    content relevant to the tenant theme
-     * @param tenantDomain tenant to which the theme is imported
-     * @throws APIManagementException if an error occurs while importing tenant theme
+     * @param themeContentInputStream content relevant to the tenant theme
+     * @param tenantDomain            tenant to which the theme is imported
+     * @throws APIManagementException if an error occurs while importing the tenant theme
+     * @throws IOException            if an error occurs while performing file or directory related operations
      */
-    public static void deployTenantTheme(InputStream themeFile, String tenantDomain) throws APIManagementException {
+    public static void importTenantTheme(InputStream themeContentInputStream, String tenantDomain)
+            throws APIManagementException, IOException {
 
         ZipInputStream zipInputStream = null;
         byte[] buffer = new byte[1024];
+        InputStream existingTenantTheme = null;
+        ByteArrayInputStream themeContent = null;
+        File tenantThemeDirectory;
+        File backupDirectory = null;
 
-        String outputFolder = "repository" + File.separator + "deployment" + File.separator + "server"
-                + File.separator + "jaggeryapps" + File.separator + "devportal" + File.separator + "site"
-                + File.separator + "public" + File.separator + "tenant_themes" + File.separator + tenantDomain;
+        int tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
 
         try {
-            //create output directory if it does not exist
-            File folder = new File(outputFolder);
-            if (!folder.exists()) {
-                if (!folder.mkdirs()) {
+            //convert InputStream to ByteArrayInputStream to be able to read the tenant theme content twice
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            IOUtils.copy(themeContentInputStream, baos);
+            byte[] bytes = baos.toByteArray();
+            themeContent = new ByteArrayInputStream(bytes);
+
+            APIAdmin apiAdmin = new APIAdminImpl();
+            //add or update the tenant theme in the database
+            if (apiAdmin.isTenantThemeExist(tenantId)) {
+                existingTenantTheme = apiAdmin.getTenantTheme(tenantId);
+                apiAdmin.updateTenantTheme(tenantId, themeContent);
+            } else {
+                apiAdmin.addTenantTheme(tenantId, themeContent);
+            }
+            //reset the InputStream to the initial position since it was completely read when adding to the database
+            themeContent.reset();
```

Comment:
Did we test this with larger files around several MBs? 
For the safe-side we can retrieve the theme from the Database without resetting and reusing the stream.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/8927/commits/b2a2a0d1edec4d4fc2ccab24e2751b55fc9b3789